include/swift/AST/DiagnosticsSIL.def

```diff
@@ -173,6 +173,9 @@ ERROR(immutable_value_passed_inout,none,
 ERROR(assignment_to_immutable_value,none,
       "immutable value '%0' may not be assigned to",
       (StringRef))
+ERROR(mutating_protocol_witness_method_on_let_constant,none,
+      "Cannot perform mutating instruction: '%0' is a 'let' constant",
+      (StringRef))
```

Comment:
@jckarter I would have preferred to use an existing string instead of creating a new one as well. Currently there are multiple diagnostics regarding improper mutation of let constants. All of those that I've found reference a specific kind of usage of the immutable value.

There are a variety of cases that are being caught by this check. These include assigning, inout usage and calling mutating methods. Depending on the Example, `assignment_to_immutable_value`, `immutable_value_passed_inout`, or one of the four variations of `mutating_method_called_on_immutable_value` would be applicable.  
Distinguishing which one is appropriate would require (most likely non-trivial) inspection of the corresponding witness method.
I decided to instead add a more universal diagnostic.

I'd be happy to adjust the wording or integrate a new, more general option into `mutating_method_called_on_immutable_value`.
If you have another idea or distinguishing the kind of mutation is a high priority for you, please let me know.